pkg/clusterapi/system.go

```diff
@@ -175,6 +176,25 @@ func (c *system) Run(ctx context.Context, installConfig *installconfig.InstallCo
 			),
 		)
 	case gcp.Name:
+		gAppCredEnvVar := "GOOGLE_APPLICATION_CREDENTIALS"
+		capgEnvVars := map[string]string{}
+		session, err := gcpic.GetSession(context.Background())
+		if err != nil {
+			return fmt.Errorf("failed to create gcp session: %w", err)
+		}
+
+		if session.Path != "" {
+			if _, err := os.Stat(session.Path); err == nil {
+				capgEnvVars[gAppCredEnvVar] = session.Path
+			}
+		}
+
+		if v, ok := capgEnvVars[gAppCredEnvVar]; ok {
+			logrus.Infof("setting %q to %s for capg infrastructure controller", gAppCredEnvVar, v)
```

Comment:
that session.Path is a valid value should have already been validated, so we should be able to trust session.Path exists. So we can simplify this a bit:

```
if session.Path != "" {
			capgEnvVars[gAppCredEnvVar] = session.Path
			logrus.Infof("setting %q to %s for capg infrastructure controller", gAppCredEnvVar, v)
		}
```
If you want to be defensive and still run the `os.Stat` again that sounds good, but let's return the error if it's not nil.